the-collab-lab/tcl-11-smart-shopping-list
# Feature(Issue #3) : Create a new shopping list with a uniquely generated token

## Description

This Pull Request (PR) adds a new `<Home/>` component to serve as a landing page. Modifies the `<List />` component to generate a new token if one doesn't exist in local storage. Refactors the `<AddItemFirebase />` and `<AddItem />` components into one `<AddItem />` component. Changes the `<AddItem />` component to add items to the firestore database based on the user token with an added `id` parameter to each item. Lastly, refactors `<AddItem />` to modify the add item form for better accessibility.

When the user selects the list link, they are redirected to the `<List/>` component view. If no token is saved into `localStorage` the user is notified their shopping list is empty and prompted with a button to add an item. Selecting the 'add item' button generates a new, unique, user token associated with a new shopping list and the user is redirected to the `<List/>` component. 

## Related Issue

Closes Issue #3 

## Acceptance Criteria


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

![Before - Empty List View](https://user-images.githubusercontent.com/32943879/90058053-36808000-dcaf-11ea-9b57-22d41a43ad6d.png)
![Before - List View](https://user-images.githubusercontent.com/32943879/90057797-e73a4f80-dcae-11ea-9796-8a703d5a3fa3.png)
![Before - Add Item View](https://user-images.githubusercontent.com/32943879/90057868-fcaf7980-dcae-11ea-8573-ac7ac1c646e0.png)

### After

![Home Page](https://user-images.githubusercontent.com/32943879/90058491-caeae280-dcaf-11ea-9bae-cd376dfabfe0.png)
![Empty Shopping List View](https://user-images.githubusercontent.com/32943879/90058498-cde5d300-dcaf-11ea-97df-f09419f1519e.png)
![Shopping List View](https://user-images.githubusercontent.com/32943879/90068652-ffb26600-dcbe-11ea-8129-fbb499c8680b.png)
![Present Token Shopping List View](https://user-images.githubusercontent.com/32943879/90058516-d3dbb400-dcaf-11ea-9736-c759c18514ce.png)
![Add Item View](https://user-images.githubusercontent.com/32943879/90068568-db568980-dcbe-11ea-8702-0e8c2448e821.png)

## Testing Steps / QA Criteria

- From your terminal, run `git checkout kl-vh-token-setup`
- Once on the `git checkout kl-vh-token-setup` branch, in the browser, visit the new `<Home/>` page (http://localhost:3000). 
- Click the `<List/>` navigation link at the bottom of the screen to be redirected to the `<List/>` component view.
- If no token is saved within `localStorage`, the message 'Your shopping list is currently empty' with an 'add new' item button is present. If a token is already saved in `localStorage` and a shopping list does exist, the list items should be rendered. 
- To create a new shopping list, click the 'add item' button. After being redirected to the `<AddItem />` component view type in a new item and select the 'add item' button. Upon submission, a new item is added to the database associated with the user token collection.
